addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/discovery/AvmUpnpDiscoveryParticipant.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.openhab.binding.avmfritz.BindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AvmUpnpDiscoveryParticipant} is responsible for discovering
+ * new and removed FRITZ!Box devices. It uses the central
+ * {@link UpnpDiscoveryService}.
+ *
+ * @author Robert Bausdorf - Initial contribution
+ */
+public class AvmUpnpDiscoveryParticipant implements
+		UpnpDiscoveryParticipant {
+	/**
+	 * Logger
+	 */
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+	/**
+	 * Provide supported thing type uid's
+	 */
+	@Override
+	public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+		return BindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS;
+	}
+
+	/**
+	 * Create a discovery result from UPNP discovery
+	 */
+	@Override
+	public DiscoveryResult createResult(RemoteDevice device) {
+		ThingUID uid = getThingUID(device);
+		if (uid != null) {
+			logger.debug("discovered: " + device.getDisplayString() + " at " 
+					+ device.getIdentity().getDescriptorURL().getHost());
+			Map<String, Object> properties = new HashMap<>();
+			properties.put(IP_ADDRESS, device.getIdentity().getDescriptorURL()
+					.getHost());
+			DiscoveryResult result = DiscoveryResultBuilder
+					.create(uid)
+					.withProperties(properties)
+					.withLabel(device.getDetails().getFriendlyName())
+					.withRepresentationProperty(IP_ADDRESS)
+					.withTTL(
+							Math.max(MIN_MAX_AGE_SECS, device.getIdentity()
+									.getMaxAgeSeconds())).build();
+			return result;
+		} else {
+			logger.debug("no uid for: " + device.getDisplayString());
+			return null;
+		}
+	}
+
+	/**
+	 * Compute a FRITZ!Box thind UID.
+	 */
+	@Override
+	public ThingUID getThingUID(RemoteDevice device) {
+		DeviceDetails details = device.getDetails();
+		if (details != null) {
+			logger.debug(details.getFriendlyName() + ": "
+					+ details.getPresentationURI());
+			ModelDetails modelDetails = details.getModelDetails();
+			if (modelDetails != null) {
+				String modelName = modelDetails.getModelName();
+				logger.debug("ModelName: " + modelName);
+				if (modelName != null) {
+					if (modelName.startsWith(BRIDGE_MODEL_NAME)) {
+						logger.debug("discovered on " + device.getIdentity().getDiscoveredOnLocalAddress());
+						return new ThingUID(BRIDGE_THING_TYPE, device
+								.getIdentity().getDescriptorURL().getHost()
+//	It world be better to use udn but in my case FB is discovered twice
```

Comment:
Hm, as I get "AVM FRITZ!Mediaserver", the UPnP discovery data seems to be different. But never mind, if you cannot reproduce this, I can further analyze and suggest a fix after the merge.